benchmark/bench_util.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include <fcntl.h>
+#include "common.h"
+#include "fileops.h"
+#include "buffer.h"
+
+static int tempdir_isvalid(const char *path)
+{
+	struct stat st;
+
+	if (p_stat(path, &st) != 0)
+		return 0;
+
+	if (!S_ISDIR(st.st_mode))
+		return 0;
+
+	return (access(path, W_OK) == 0);
+}
+
+int gitbench__create_tempdir(char **out)
+{
+	git_buf buf_env = GIT_BUF_INIT;
+	git_buf tempdir = GIT_BUF_INIT;
+	static const char *tempname = "libgit2_bench_XXXXXX";
+	int error = -1;
+
+	if ((git_buf_sets(&buf_env, getenv("BM_TEMP")) < 0) ||
```

Comment:
We could do what clar does here and read from `BM_TEMP` but then fall back on to the standard variables if it's not set, rather than failing.